sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Triggers.scala

```diff
@@ -31,8 +31,9 @@ private object Triggers {
 
   def convert(interval: String): Long = {
     val cal = CalendarInterval.fromCaseInsensitiveString(interval)
-    if (cal.months > 0) {
-      throw new IllegalArgumentException(s"Doesn't support month or year interval: $interval")
+    if (cal.months > 0 || cal.days > 0) {
+      throw new IllegalArgumentException(
+        s"Doesn't support day, week, month or year interval: $interval")
```

Comment:
This will fail a current valid usage: `Trigger.ProcessingTime("10 day")`
Maybe we should still keep converting day interval and use a 24 hours = 1 day assumption